bcdaworker/main.go

```diff
@@ -293,5 +295,61 @@ func main() {
 	fmt.Println("Starting bcdaworker...")
 	workerPool := setupQueue()
 	defer workerPool.Close()
+	expiredScheduler()
 	waitForSig()
 }
+
+func updateStatus(hrThreshold int) {
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	var jobs []models.Job
+	err := db.Find(&jobs, "status = ?", "Completed").Error
+	if err != nil {
+		log.Error(err)
+	}
+
+	expDir := os.Getenv("FHIR_EXPIRED_DIR")
+	if _, err = os.Stat(expDir); os.IsNotExist(err) {
+		err = os.MkdirAll(expDir, os.ModePerm)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
+	for _, j := range jobs {
+		t := j.CreatedAt
+		elapsed := time.Since(t).Hours()
+		if int(elapsed) >= hrThreshold {
+			j.Status = "Expired"
+			err = db.Save(j).Error
```

Comment:
This isn't atomic or robust in the face of failures: If this code dies halfway through, you will have marked the job as expired in the DB but left the files orphaned on the disk.

Reply:
really good point! im thinking about how to refactor this to be atomic but if you have any suggestions let me know.